storage-node/src/services/sync/acceptPendingObjects.ts

```diff
@@ -54,61 +59,87 @@ export class AcceptPendingObjectsService {
         bucketKeyPairs,
         uploadBuckets
       )
-      await this.instance.loadPendingDataObjects()
       this.instance.runWithInterval(api, workerId, maxTxBatchSize, intervalMs)
     }
     return this.instance
   }
 
-  private runWithInterval(api: ApiPromise, workerId: number, maxTxBatchSize: number, intervalMs: number) {
-    const run = () => {
-      this.acceptPendingDataObjects(api, workerId, maxTxBatchSize)
-        .catch((err) => logger.error(`Failed to register pending data objects as accepted in runtime: ${err}`))
-        .finally(() => setTimeout(run, intervalMs))
-    }
-    run()
+  pendingObjectExists(id: string): boolean {
+    return fs.existsSync(path.join(this.pendingDataObjectsDir, id))
+  }
+
+  private async getPendingObjectsFromFolder(): Promise<string[]> {
+    const dirEntries = await fsPromises.readdir(this.pendingDataObjectsDir, { withFileTypes: true })
+    return dirEntries.filter((entry) => entry.isFile()).map((entry) => entry.name)
   }
 
-  getPendingDataObject(objectId: string): [string, string] | undefined {
-    return this.pendingDataObjects.get(objectId)
+  private runWithInterval(api: ApiPromise, workerId: number, maxTxBatchSize: number, intervalMs: number) {
+    ;(async () => {
+      for (;;) {
+        try {
+          const ids = await this.getPendingObjectsFromFolder()
+          const objectsToAccept = await this.processPendingObjects(ids)
+          await this.acceptPendingDataObjects(api, workerId, objectsToAccept, maxTxBatchSize)
+          await new Promise((resolve) => setTimeout(resolve, intervalMs))
+        } catch (err) {
+          logger.error(`Error processing pending objects: ${err}`)
+        }
+      }
+    })().catch((err) => {
+      logger.error(`Accept pending objects service died: ${err}`)
+    })
```

Comment:
```suggestion
    })()
```
Since the async function is in `try...catch`, I don't think any promise rejections will ever be caught in `.catch` function. WDYT?

Reply:
You are correct, I think I was forced to add a catch handler because typescript complained. I'll try to use the previous code style to make this whole function cleaner. The reason I ended up with this was this was the point where I was debugging the issue which ended up being the invalid spread operator on the extrinsic result.